### PR TITLE
Jonavin keymap kastenwagen48

### PR DESCRIPTION
**Pull request by Jonavin**
## Description

Jonavin Kastenwagen 48 Keymap
Designed to match my other keymaps

NOTE: Uses userpace user/jonavin

Feature changes from Default
  - Bottom row mapping LCtrl, LAlt, LGui, spacebars, MO(2)/LOWER, MO(1)/FN, RCtrl
  - CAPS as MO(3)/RAISE - tap to toggle
  - numpad in MO(3) layer
  - double tap L Shift for CAPS
  - RESET moved to under encoder in MO(2)
  - implement Win Key Lock 
  - GRV Escape overides
  - ENCODERS
    - TOP LEFT
      - Volume up/down
    - TOP RIGHT
      - Left Shift held - change layers
      - Right Shift held - Page Up/Down
      - Left Ctrl held - navigate next/prev words
      - Left Alt held - change media next/prev track
      - RAISE + Encode push toggles between volume change and tab scrolling
    - ENCODER_DIRECTION_FLIP has been defined to avoid issues with encoder going in opposition direction

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/nopunin10did/kastenwagen48/keymaps/jonavin/kastenwagen.json [hunk 1]
```diff
@@ -0,0 +1,72 @@
+{
```
Comment (drashna) [inline keyboards/nopunin10did/kastenwagen48/keymaps/jonavin/kastenwagen.json:1]: This appears to be a VIA json. 
https://docs.qmk.fm/#/pr_checklist?id=keyboard-prs

> Do not include VIA json files in the PR. These do not belong in the QMK repository as they are not used by QMK firmware – they belong in the [VIA Keyboard Repo](https://github.com/the-via/keyboards>

File: keyboards/nopunin10did/kastenwagen48/keymaps/jonavin/keymap.c [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        switch (index) {
+            case 0:         // Top left encoder
+                encoder_action_volume(clockwise);
+                break;
+            default:
+                break;
+        }
+        return true;  // fall to encoder_update_user and encoder_update_kb definitions
+    }
+#endif
```
Comment (drashna) [inline keyboards/nopunin10did/kastenwagen48/keymaps/jonavin/keymap.c:67]:
```suggestion
#ifdef ENCODER_ENABLE       // Encoder Functionality
bool encoder_update_keymap(uint8_t index, bool clockwise) {
    switch (index) {
        case 0:         // Top left encoder
            encoder_action_volume(clockwise);
            break;
        default:
            break;
    }
    return true;  // fall to encoder_update_user and encoder_update_kb definitions
}
#endif
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/nopunin10did/kastenwagen48/keymaps/jonavin/kastenwagen.json (hunk 1 above)
Comment (Jonavin) [reply keyboards/nopunin10did/kastenwagen48/keymaps/jonavin/kastenwagen.json]: It was in the .ignore file but somehow got into this  pr. It’s been removed from the folder.

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**